test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommandTests.cs

```diff
@@ -2135,7 +2135,8 @@ public async Task RestoreCommand_RestoreFloatingVersionWithIgnoreFailingLocalSou
                 }");
 
                 var specPath = Path.Combine(projectDir, "TestProject", "project.json");
-                var spec = JsonPackageSpecReader.GetPackageSpec(configJson.ToString(), "TestProject", specPath);
+                var spec = JsonPackageSpecReader.GetPackageSpec(configJson.ToString(), "TestProject", specPath)
+                    .EnsureProjectJsonRestoreMetadata();
```

Comment:
> .EnsureProjectJsonRestoreMetadata(); [](http://example.com/codeflow?start=19&length=37)

NIT: Could we have these on the same line as above? They look odd on their own line.

Reply:
I disagree for multiple reasons. I'm sure this response is over the top, but here it is anyway.

Firstly, this type of "method chaining" is particularly common for "fluent interfaces" which is common/popular enough to have its own wikipedia article: https://en.wikipedia.org/wiki/Fluent_interface
Most of the examples have each chained method on a different line.

Here's an example where putting each chained method on a new line makes it easier to see that package id, version, and icon is being "built". If it was one line line, it wouldn't be so obvious how many different metadata items are being set:
https://github.com/NuGet/NuGet.Client/blob/3dbfac4cf9633d012ff22536cdb895e2f4e4b487/test/TestUtilities/Test.Utility/Protocol/LocalPackageSearchMetadataFixture.cs#L24-L28

Secondly, consider other scenarios where method chaining is used, but with nested expressions. Look at this parenthesis soup (I count 7 pairs): https://github.com/NuGet/NuGet.Client/blob/3dbfac4cf9633d012ff22536cdb895e2f4e4b487/test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Services/NuGetPackageSearchServiceTests.cs#L274
Maybe this is just a bad example, and we shouldn't say a technique/style is bad just because of some bad examples, but what is the final `Build()` being called on? The `NuGetVersion.Parse` result? On the `new PackageIdentity` instance? On the result of `PackageSearchMetadataBuilder.FromIdentity`? On `Task.FromResult`? It would be so much easier to understand if it was on multiple lines and indented appropriately, so we could see which methods are being chained to what. Of course, there are multiple ways this could have been written differently to only use 2 to 4 short lines. However, if you search the repo for `))).`, I see 138 instances, so it appears that chaining & nesting method calls isn't uncommon in our code, or 1540 instances of `)).`. LINQ really makes it more common, since most LINQ methods take a delegate as a parameter, and it's very common to call methods within an anonymous method. But language features added to C# in the last 5 years can help if we can get people in the team to use new patterns, rather than copying the existing code. Anyway, my point is that chained methods on fluent interfaces on new lines makes understanding nested calls almost trivial, although it does need to be indented correctly.

Finally, there are people who think that having an [enforced max line length](https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#max_line_length) is a good idea. For better or worse, [dotnet's format style rules doesn't support it](https://github.com/dotnet/format/blob/main/docs/Supported-.editorconfig-options.md?plain=1#L13), but [resharper has had its own setting for years](https://learn.microsoft.com/en-us/answers/questions/1007988/line-length-enforcement-for-visual-studio-and-dotn), which the team I worked for before joining Microsoft used.

Similarly, while I don't, some people code on a monitor in vertical orientation, significantly reducing the horizontal space. Needing to scroll left and right is much more harmful to readability. There are definitely times I'm tempted to switch to vertical, because most code is not so wide, and having more lines would be better.

And last note on this, the hardware provided to me when I go to the office is limited to 1920x1080. Even if I got a higher resolution monitor, I'd still need a laptop docking station capable of driving it. I'm not going to write lines that don't fit on the screen I'm writing it on. Although most of these examples in this PR would JUST fit on a single line (considering I have Solution Explorer open, eating some horizontal space). Unless I switch to vertical monitor orientation, in which case it wouldn't have a chance to fit.

If you made it this far, thank you for reading my essay.

Reply:
It was only a nit @zivkan, dang :D